ipfs-lite/ipfs_lite.proto

```diff
@@ -0,0 +1,218 @@
+syntax = "proto3";
+
+import "google/protobuf/any.proto";
+
+option java_multiple_files = true;
+option java_package = "io.textile.grpc_ipfs_lite";
+option java_outer_classname = "IpfsLiteProto";
+
+// Model messages
+
+message AddParams {
+    string layout = 1;
+    string chunker = 2;
+    bool rawLeaves = 3;
+    bool hidden = 4;
+    bool shared = 5;
+    bool noCopy = 6;
+    string hashFun = 7;
+}
+
+message Block {
+    bytes rawData = 1;
+    string cid = 2;
+}
+
+message Link {
+    string name = 1;
+    int64 size = 2;
+    string cid  = 3;
+}
+
+message NodeStat {
+    string hash = 1;
+    int32 numLinks = 2;
+    int32 blockSize = 3;
+    int32 linksSize = 4;
+    int32 dataSize = 5;
+    int32 cumulativeSize = 6;
+}
+
+message Node {
+    Block block = 1;
+    repeated Link links = 2;
+    NodeStat state = 3;
+    int64 size = 4;
+}
+
+// Request and Response messages
+
+message AddFileRequest {
+    oneof msgtype {
+        AddParams addParams = 1;
+        bytes chunk = 2;
+    }
+}
+
+message AddFileResponse {
+    Node node = 1;
+}
+
+message GetFileRequest {
+    string cid = 1;
+}
+
+message GetFileResponse {
+    bytes data = 1;
+}
+
+message GetNodeRequest {
+    string cid = 1;
+}
+
+message GetNodeResponse {
+    Node node = 1;
+}
+
+message GetNodesRequest {
+    repeated string cid = 1;
+}
+
+message GetNodesResponse {
+    oneof option {
+        Node node = 1;
+        string error = 2;
+    }
+}
+
+message ResolveLinkRequest {
+    string nodeCid = 1;
+    repeated string path = 2;
+}
+
+message ResolveLinkResponse {
+    Link link = 1;
+    repeated string remainingPath = 2;
+}
+
+message ResolveRequest {
+    string nodeCid = 1;
+    repeated string path = 2;
+}
+
+message ResolveResponse {
+    google.protobuf.Any object = 1;
```

Comment:
Not sure about how to model this. In go, it looks like:
```
// Resolve resolves a path through this node, stopping at any link boundary
// and returning the object found as well as the remaining path to traverse
Resolve(path []string) (interface{}, []string, error)
```
What does that `interface{}` return type mean and how would you deal with that in go?

Reply:
On the go-side you'd have some context about the expected object such that you could do a type check. `Any` is the right way for `interface{}`, though still not great: https://github.com/golang/protobuf/issues/272#issuecomment-272231323